Allow to reuse an existing session by fixing a bug in session comparison.
See http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=573258

--- a/sesman/session.c
+++ b/sesman/session.c
@@ -55,6 +55,19 @@ session_get_bydata(char* name, int width, int height, int bpp, int type)
 {
   struct session_chain* tmp;
 
+  /* convert from SCP_SESSION_TYPE namespace to SESMAN_SESSION_TYPE namespace */
+  switch (type)
+  {
+    case SCP_SESSION_TYPE_XVNC:
+      type = SESMAN_SESSION_TYPE_XVNC;
+      break;
+    case SCP_SESSION_TYPE_XRDP:
+      type = SESMAN_SESSION_TYPE_XRDP;
+      break;
+    default:
+      return 0;
+  }
+
   /*THREAD-FIX require chain lock */
   lock_chain_acquire();
 
